### PR TITLE
added Sierra template save/load GUIs, Ctrl+Q uses custom GUI now

### PR DESCRIPTION
**Pull request by khrismuc**
Blue Save and Load GUIs from the Sierra template got re-added, Ctrl+Q keypress now shows custom quit GUI instead of built-in one

### CODE REVIEW

**Comment by morganwillcock** [comment]
Thanks. One issue when I've tested this: when the quit game dialog is opened, if the verbcoin GUI is already open it remains active and clickable.

One of the goals of updating the BASS template was to allow the allow the pause of the main interface GUI to be optional (so the user could choose whether the game keeps running or not when the mouse goes to the top of the screen), so it is probably best to consider the same thing here and decide what to do when the verbcoin GUI is open but not blocking the game. That said, the BASS template is also missing the same save and load GUIs and the keyboard shortcut to quit, so it is likely that I will try to apply and check the same changes to both templates.

So in summary, if you are OK to leave this open for a little while I'll try to put the same changes into the BASS template and check that any pause/close behaviour of the GUIs is consistent.

**Comment by khrismuc** [comment]
Hey, I fixed the issue you mentioned (Ctrl+Q shows Quit GUI and hides Verb Coin if it's open), I also added a CleanUp() function that takes care of the DynamicSprite (previously, ending the game caused a warning).

I'm of course fine with leaving this open; I was wary about it anyway because people should ideally test my code thoroughly before it is added to anything public ;)
I pushed the changes to my repo and can see they also ended up in the pull request.

**Comment by morganwillcock** [comment]
I've had another look and started to add the equivalent into the BASS template too, it has highlighted an issue that since both the BASS top bar and the verbcoin can be used in a pausing or unpausing styles they cannot be preconfigured to disable themselves when the game is paused, which means that when the load or save GUIs are opened the BASS or verbcoin GUIs may be open and active. It is more of an issue on the BASS template if you wanted a demo: https://github.com/morganwillcock/ags-template-source/tree/load-save

I'm not sure that forcing everything to close when other GUIs open is a good idea and it conflicts with using escape as a 'close button' for other purposes, so I think I will add a new configuration option to both script modules where you can specify other GUIs which should be yielded to (so BASS and verbcoin GUIs can then choose not to open or to disable themselves if other specified GUIs are visible). Maybe I am overthinking things though, and maybe you have a better suggestion...

**Comment by morganwillcock** [comment]
I think I've got it working on both templates whilst still keeping the logic of the script modules separate: https://github.com/morganwillcock/ags-template-source/tree/load-save

I'll test it a little more, but this should let the load, save, and quit GUIs work everywhere.

**Comment by ivan-mogilko** [comment]
Sorry, have to poke, because there's another user on boards asking about this.

**Comment by morganwillcock** [comment]
I'll merge this today or tomorrow (just need to re-test to convince myself I didn't break anything). I also noticed the cursor is not set for dialog selection, so I will fix that too (separately).

**Comment by morganwillcock** [comment]
I've used the original commits plus additional changes on top, and everything seems OK.
